### ISSUE TITLE
[BUG] Custom port for cloning

### ISSUE THREAD

**Issue by Nate-Wilkins** [zeapo/Android-Password-Store#2264]
### Describe the bug

When cloning or configuring a remote repository entering a port into the `Repository URL` text box the clone doesn't use the provided port.

### Steps to reproduce

1. Go to the app
2. Attempt to clone a repository
3. Type in a valid git URL with a custom port number other than 22
4. Notice that the clone failed and indicates that it tried to connect to port 22

### Expected behavior

I was expecting this to be the same as any git clone on a linux machine where putting in a `-p ...` would augment the request.

### Device information

- Device: [Samsung Note 9]
- OS: [Android v10]
- App version [v1.13.5]

### Additional context

When I was using the legacy app this worked fine.

I only recently stopped using the legacy version because I could no longer download with a custom port. 

I've always had issues with this though which is partly why I'm filling this bug report :) 

Love the work you guys have done.
If you need additional help I could dig into the source code.

Thanks!

**Comment by msfjarvis** [comment]
I just tried this out and cannot reproduce the issue, custom ports work just fine.

I used [soft-serve](https://github.com/charmbracelet/soft-serve) to spin up a temporary Git server on port 23231 of my computer, then cloned the [pass-test](https://msfjarvis.dev/apsg/pass-test) repository with the URL `ssh://admin@192.168.1.2:23231/pass-test` and got a successful clone over SSH.

**Comment by Nate-Wilkins** [comment]
Hmm.

These were my settings:

```
Repository URL:
git@my.domain.com:3033nate-wilkins/password-store.git

Branch:
master

Authentication Mode:
SSH Key
```

And get the following error:

```
An error occurred during a Git operation

failed to connect to my.domain.com/76.30.10.15 (port 22) from /:: (port 36746): connect failed ECONNREFUSED (Connection refused)
```

What's strange is the `(port 22)` and the other `(port 36746)` because I don't have any configuration in which either of those ports are used? I mean I might be missing something but it does seem strange.

**Comment by msfjarvis** [comment]
Can you try the [snapshot build](https://docs.passwordstore.app/docs/users/release-channels/)? Bear in mind that it no longer uses OpenKeychain (#1523) so you'll have to import your GPG private key directly in the app's settings to use it.

**Comment by Nate-Wilkins** [comment]
@msfjarvis I haven't tried the snapshot build but I'll have to try that next. 

However, I was able to get past this error message by prefixing the clone url with `ssh://` and replacing the `:` before the repository path with a `/`. So it seems that its now connecting to the remote repository but now I'm getting the following:

```
An error occurred during a Git operation
Your local repository appears to be an incomplete Git clone, please delete and re-clone from settings
```

Even though I deleted the local `.password-store` repository... 

Long story short I solved this by cloning through another tool and pointing PasswordStore to the local repository.

**Comment by AnrDaemon** [comment]
This MUST be clearly documented inside the app itself.
I've tried for weeks to access my repo (which is behind the ingress server) on a custom forwarded port. And always get the "socket closed" error that did not tell much of anything.

**Comment by AnrDaemon** [comment]
To clarify: using URL scheme worked at least past the "socket error" response to the point where I've got a password prompt.
Now I have to double check the SSH keys setup, but it SEEMS to be working on connection level.
